app/views/EditAry/AssessmentRegistryForm/AdditionalDocument/FileUpload/AryFileUpload/index.tsx

```diff
@@ -0,0 +1,79 @@
+import React, { useCallback } from 'react';
+import {
+    FileInput,
+} from '@the-deep/deep-ui';
+import { IoCloudUpload } from 'react-icons/io5';
+import { AdditionalDocumentType, AssessmentRegistryDocumentTypeEnum } from '#generated/types';
+import { useLazyRequest } from '#base/utils/restRequest';
+
+import styles from './styles.css';
+
+interface Props {
+    onSuccess: (
+        v: AdditionalDocumentType,
+        documentType: AssessmentRegistryDocumentTypeEnum,
+    ) => void;
+    name: AssessmentRegistryDocumentTypeEnum;
+    acceptFileType?: string;
+}
+
+interface UploadType {
+    title: string;
+    file: File;
+    isPublic: boolean;
+}
+
+function AryFileUpload(props: Props) {
+    const {
+        onSuccess,
+        name,
+        acceptFileType,
+    } = props;
+
+    const {
+        pending,
+        trigger,
+    } = useLazyRequest<AdditionalDocumentType, UploadType | undefined | null>({
+        url: 'server://files/',
+        method: 'POST',
+        formData: true,
+        body: (ctx) => ctx,
+        onSuccess: (response) => {
+            onSuccess(response, name);
+        },
+        onFailure: (err) => {
+            console.log(err);
+        },
+        failureMessage: 'Upload failed.',
+    });
```

Comment:
Let's discuss with the backend team to create a gql endpoint for this. We can skip for now, but good to have in future.